server/executor/poller_executor.go

```diff
@@ -98,17 +102,46 @@ func (pe DefaultPollerExecutor) ExecuteRequest(request *PollingRequest) (bool, m
 		return false, model.Run{}, err
 	}
 
+	if request.IsFirstRequest() {
+		connectionResult := traceDB.TestConnection(request.ctx)
+
+		err = pe.eventEmitter.Emit(request.ctx, events.TraceDataStoreConnectionInfo(request.test.ID, request.run.ID, connectionResult))
+		if err != nil {
+			log.Printf("[PollerExecutor] Test %s Run %d: failed to emit TraceDataStoreConnectionInfo event: error: %s\n", request.test.ID, request.run.ID, err.Error())
+		}
+	}
+
+	err = pe.eventEmitter.Emit(request.ctx, events.TraceFetchingStart(request.test.ID, request.run.ID))
+	if err != nil {
+		log.Printf("[PollerExecutor] Test %s Run %d: failed to emit TraceFetchingStart event: error: %s\n", request.test.ID, request.run.ID, err.Error())
+	}
+
 	traceID := run.TraceID.String()
 	trace, err := traceDB.GetTraceByID(request.ctx, traceID)
 	if err != nil {
+		anotherErr := pe.eventEmitter.Emit(request.ctx, events.TraceFetchingError(request.test.ID, request.run.ID, err))
+		if anotherErr != nil {
+			log.Printf("[PollerExecutor] Test %s Run %d: failed to emit TraceFetchingError event: error: %s\n", request.test.ID, request.run.ID, anotherErr.Error())
+		}
+
```

Comment:
A would expect a connection test to happen here, so if it fails, we at least know if the data store connection is working. We might want to filter out the `ErrTraceNotFound` error though.

Reply:
Have you considered this connection a complement of the `TraceFetchingError` event? Something like this:
```go
if err != nil {
		connectionResult := traceDB.TestConnection(request.ctx)

		anotherErr := pe.eventEmitter.Emit(request.ctx, events.TraceFetchingError(request.test.ID, request.run.ID,  connectionResult, err))
		if anotherErr != nil {
			log.Printf("[PollerExecutor] Test %s Run %d: failed to emit TraceFetchingError event: error: %s\n", request.test.ID, request.run.ID, anotherErr.Error())
		}
}
```